src/execution/values.ts

```diff
@@ -186,37 +208,41 @@ export function getArgumentValues(
       const variableName = valueNode.name.value;
       if (
         variableValues == null ||
-        !hasOwnProperty(variableValues, variableName)
+        variableValues.coerced[variableName] === undefined
```

Comment:
Hmmm.

The existing pre-PR version of `getArgumentValues` seems to assume that we could have a coerced value of `undefined` as evidenced by the previous use of `hasOwnProperty` that you are highlighting....

But the existing pre-PR version of `coerceInputValue` would throw errors if parsing actually led to undefined.

https://github.com/graphql/graphql-js/blob/76e47fcd7d432f515d5bb99e95af1851148a0c54/src/utilities/coerceInputValue.ts#L170-L176

So I don't believe we actually previously supported coerced values of `undefined`?

Reply:
Tagging @leebyron  as per request at last WG

Reply:
After further review, my interpretation is still that this code change is non-breaking and unrelated to this PR => I can extract it to a separate PR.

The changed code block operated on the post-coerced "internal" variable values for which `undefined` has never been a valid value.

The only other significant use of `hasOwnProperty` is within `coerceVariableValues` where it is used on pre-coerced "external" values that could potentially have the value `undefined`:

https://github.com/graphql/graphql-js/blob/76e47fcd7d432f515d5bb99e95af1851148a0c54/src/execution/values.ts#L95-L107

Although the use of `hasOwnProperty` may lead to some interesting behavior, see #2533

Reply:
Actually, I think #2533 can be closed and the existing code is more accurate as there really never should be a post-coercion value of `undefined`.

Reply:
After continued further review, I think this code change (from a `!hasOwnProperty(...)` check to a simply `=== undefined` should be reverted.

To be precise, within the existing code base `coerceInputValue` can yield a result of undefined, but will trigger an error, and when `execute` uses => `getVariableValues`  => `coerceVariableValues` => `coerceInputValue`, it will abort execution if any coercion errors are encountered, such that `getArgumentValues` need not worry whether there is a value of `undefined` or the value is not present. But these library functions are exported, and so we should take care to distinguish between them....